### PR TITLE
PR for adding additional features for Wikibase

### PR DESCRIPTION
**Pull request by GlorianY**
PR for adding additional features for Wikibase

### CODE REVIEW

File: revscoring/features/wikibase/datasources/revision_oriented.py [hunk 1]
```diff
@@ -104,50 +144,119 @@ def __init__(self, name, revision_datasources):
 
 
 def _process_item_doc(text):
-    if text is not None:
-        return json.loads(text)
-    else:
-        return None
+	if text is not None:
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: Use tabs and not spaces.

File: revscoring/features/wikibase/datasources/revision_oriented.py [hunk 2]
```diff
@@ -75,7 +76,7 @@ def __init__(self, name, revision_datasources):
             name + ".sources", _process_sources, depends_on=[self.item]
         )
         """
-        A `set` of unique sources in the revision
+        A `set` of claim which contains sources in the revision
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: Breaks backwards compatibility.

File: revscoring/features/wikibase/datasources/excluded_qids.csv [hunk 3]
```diff
@@ -0,0 +1,477 @@
+328,English Wikipedia
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/excluded_qids.csv]: This is Wikidata specific and probably shouldn't be included in a wikibase-general revscoring.  Maybe we should move this into wikiclass.

File: revscoring/features/wikibase/datasources/revision_oriented.py [hunk 4]
```diff
@@ -104,50 +144,119 @@ def __init__(self, name, revision_datasources):
...
+	
+	return set(
+		(property, _claim_to_str(claim), i) 
+		for property in item.claims 
+		for claim in item.claims[property] 
+		for i, source in enumerate(claim.sources) 
+		
+	)
+
+def _process_external_sources_ratio(item):
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: This function is way too large.  It should have logic broken up more.

File: revscoring/features/wikibase/datasources/revision_oriented.py [hunk 5]
```diff
@@ -159,13 +268,85 @@ def _process_labels(item):
...
 def _process_sitelinks(item):
-    return item.sitelinks
+	return item.sitelinks
 
 
 def _process_descriptions(item):
     return item.descriptions
 
-
+def _process_complete_translations(item):
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: This is also wikidata specific and should probably live in wikiclass.

File: revscoring/features/wikibase/datasources/revision_oriented.py [hunk 6]
```diff
@@ -175,3 +356,11 @@ def _claim_to_str(claim):
         return repr(claim.target)
     else:
         return str(claim.target)
+
+#read CSV, line by line
+def read_file(file_object):
```
Comment (halfak) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: This function doesn't do anything useful.  

You can just do:
```
for line in file_object:
    ...
```

File: revscoring/features/wikibase/datasources/revision_oriented.py (hunk 1 above)
Comment (Ladsgroup) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: I think Aaron meant the exact opposite. Just to be clear.

Comment (GlorianY) [inline revscoring/features/wikibase/datasources/revision_oriented.py]: So use spaces instead of tabs right?

**Pull request closed without merging**